platform/asterix/boot/src/drivers/flash/mx25u.c

```diff
@@ -1,81 +1,114 @@
-#include <drivers/flash.h>
-#include <drivers/dbgserial.h>
-
 #include <board.h>
-
+#include <drivers/dbgserial.h>
+#include <drivers/flash.h>
 #include <nrfx_qspi.h>
-
-static nrfx_qspi_config_t config =
-	NRFX_QSPI_DEFAULT_CONFIG(BOARD_QSPI_SCK_PIN, BOARD_QSPI_SCN_PIN, BOARD_QSPI_IO0_PIN,
-				 BOARD_QSPI_IO1_PIN, BOARD_QSPI_IO2_PIN, BOARD_QSPI_IO3_PIN);
+#include <system/passert.h>
 
 #define SPI_NOR_CMD_RDID 0x9F
 #define SPI_NOR_CMD_RDPD 0xAB
+#define SPI_NOR_CMD_ENRST 0x66
+#define SPI_NOR_CMD_RST 0x99
+#define SPI_NOR_CMD_EN4B 0xB7
+#define SPI_NOR_CMD_RDSR1 0x05
+#define SPI_NOR_CMD_RDSR2 0x35
+#define SPI_NOR_CMD_WRSR 0x01
+
+static void prv_read_register(uint8_t instruction, uint8_t *data, uint32_t length) {
+  nrfx_err_t err;
+  nrf_qspi_cinstr_conf_t instr = NRFX_QSPI_DEFAULT_CINSTR(instruction, length + 1);
+  instr.io2_level = true;
+  instr.io3_level = true;
+  err = nrfx_qspi_cinstr_xfer(&instr, NULL, data);
+  PBL_ASSERTN(err == NRFX_SUCCESS);
+}
+
+static void prv_write_register(uint8_t instruction, const uint8_t *data, uint32_t length) {
+  nrfx_err_t err;
+  nrf_qspi_cinstr_conf_t instr = NRFX_QSPI_DEFAULT_CINSTR(instruction, length + 1);
+  instr.io2_level = true;
+  instr.io3_level = true;
+  err = nrfx_qspi_cinstr_xfer(&instr, data, NULL);
+  PBL_ASSERTN(err == NRFX_SUCCESS);
+}
+
+void flash_init(void) {
+  uint8_t sr[2];
+  nrfx_qspi_config_t config =
+      NRFX_QSPI_DEFAULT_CONFIG(BOARD_QSPI_SCK_PIN, BOARD_QSPI_SCN_PIN, BOARD_QSPI_IO0_PIN,
+                               BOARD_QSPI_IO1_PIN, BOARD_QSPI_IO2_PIN, BOARD_QSPI_IO3_PIN);
+
+  config.prot_if.readoc = NRF_QSPI_READOC_READ4IO;
+  config.prot_if.writeoc = NRF_QSPI_WRITEOC_PP4O;
+  config.phy_if.sck_freq = NRF_QSPI_FREQ_DIV4;
```

Comment:
nit: elsewhere, we spell this `NRF_QSPI_FREQ_32MDIV4`

Reply:
actually, it's pebbleos driver that needs update

https://github.com/NordicSemiconductor/nrfx/blob/bacc23e5387964462e5638f0c914c2ab5802fa3c/hal/nrf_qspi.h#L119-L158